contracts/util/LoanLib.sol

```diff
@@ -0,0 +1,339 @@
+        @return uint256 The minimum collateral value threshold required.
+     */
+    function getCollateralNeededInTokens(
+        TellerCommon.Loan storage loan,
+        SettingsInterface settings
+    ) public view returns (int256 neededInLendingTokens, uint256 escrowLoanValue) {
+        if (!isActiveOrSet(loan) || loan.loanTerms.collateralRatio == 0) {
+            return (0, 0);
+        }
+
+        /*
+            The collateral to principal owed ratio is the sum of:
+                * collateral buffer percent
+                * loan interest rate
+                * liquidation reward percent
+                * X factor of additional collateral
+        */
+        // To take out a loan (if status == TermsSet), the required collateral is (principal owed * the collateral ratio).
+        uint256 requiredRatio = loan.loanTerms.collateralRatio;
+        neededInLendingTokens = int256(loan.principalOwed);
+
+        // For the loan to not be liquidated (when status == Active), the minimum collateral is (principal owed * X collateral factor).
+        // If the loan has an escrow account, the minimum collateral is ((principal owed - escrow value) * X collateral factor).
+        if (loan.status == TellerCommon.LoanStatus.Active) {
+            uint256 bufferPercent = settings.getPlatformSettingValue(
+                settings.consts().COLLATERAL_BUFFER_SETTING()
+            );
+            uint256 liquidateEthPrice = settings.getPlatformSettingValue(
+                settings.consts().LIQUIDATE_ETH_PRICE_SETTING()
+            );
+            requiredRatio = requiredRatio
+                .sub(loan.loanTerms.interestRate)
+                .sub(bufferPercent)
+                .sub(liquidateEthPrice.diffOneHundredPercent());
+
+            if (loan.escrow != address(0)) {
+                escrowLoanValue = EscrowInterface(loan.escrow).calculateLoanValue();
+                neededInLendingTokens -= int256(escrowLoanValue);
+            }
+        }
+
+        neededInLendingTokens = neededInLendingTokens.percent(requiredRatio);
+    }
+
+    /**
+        @notice It gets the current liquidation info for a given loan.
+        @param loan The loan to get the info.
+        @param loansContract The loans contract instance for the loan.
+        @return liquidationInfo get current liquidation info for the given loan id.
+     */
+    function getLiquidationInfo(
+        TellerCommon.Loan storage loan,
+        LoansInterface loansContract
+    ) public view returns (TellerCommon.LoanLiquidationInfo memory liquidationInfo) {
+        liquidationInfo.collateralInfo = getCollateralInfo(loan, loansContract);
+        liquidationInfo.amountToLiquidate = getTotalOwed(loan);
+
+        // Maximum reward is the calculated value of required collateral minus the principal owed (see LoanLib.getCollateralNeededInTokens).
+        int256 maxReward = liquidationInfo.collateralInfo.neededInLendingTokens -
```

Comment:
why we are using int256 (instead of uint256)? Please, use `value.sub(anotherValue)` instead of `-`

Reply:
We spend a good minute (couple hours) going over this. Basically the reward and the required collateral is adjusted based on the value of the escrow. 

The equation for being able to liquidate is:
`canLiquidate(collateralAmount) = collateralValue < owed + liquidation reward - escrow value`

The max liquidation reward is 5% of what the liquidator pays on whats owed for the loan. Now, if the value of the collateral and escrow drop below the 5% value, we cannot pay the liquidator the 5%. 
AND 
If the value of the collateral and escrow continue to drop, there could potentially be a negative reward amount. A.K.A. the liquidator pays the loan and receives less value than he paid. Unlikely to happen, but possible.

Same thing with the `collateralInfo.neededInLendingTokens`. If the collateral and escrow value rise in value, you are not required to have as much locked in collateral. If the collateral and escrow grow past a certain point, the `neededInLendingTokens` can become negative. That basically means that the borrower can withdraw the full collateral value but nothing else.